### PR TITLE
COT traits

### PR DESCRIPTION
**Pull request by sinui0**
This PR adds COT traits and implements ideal functionalities for them.

This will support the incoming Ferret COT implementation.

### CODE REVIEW

File: ot/mpz-ot/src/ideal/shared/cot.rs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+mod tests {
+    use itybity::IntoBits;
+    use rand::Rng;
+    use rand_chacha::ChaCha12Rng;
+    use rand_core::SeedableRng;
+
+    use super::*;
+
+    #[tokio::test]
+    async fn test_ideal_ot() {
```
Comment (th4s) [inline ot/mpz-ot/src/ideal/shared/cot.rs]: Better name `test_ideal_cot_shared`

**Review by th4s — APPROVED** [review]
:+1:

File: ot/mpz-ot/src/lib.rs [hunk 2]
```diff
@@ -68,6 +68,30 @@ where
     ) -> Result<(), OTError>;
 }
 
+/// A correlated oblivious transfer sender.
+#[async_trait]
+pub trait COTSender<T>: ProtocolMessage
+where
+    T: Send + Sync,
+{
+    /// Obliviously transfers the correlations to the receiver.
```
Comment (themighty1) [inline ot/mpz-ot/src/lib.rs]: I changed "correlation" to "correlated messages" for consistency and clarity.
```suggestion
    /// Obliviously transfers the correlated messages to the receiver.
```

File: ot/mpz-ot/src/lib.rs [hunk 3]
```diff
@@ -68,6 +68,30 @@ where
...
+    ///
+    /// # Arguments
+    ///
+    /// * `sink` - The IO sink to the receiver.
+    /// * `stream` - The IO stream from the receiver.
+    /// * `msgs` - The `0`-bit messages to obliviously transfer.
```
Comment (themighty1) [inline ot/mpz-ot/src/lib.rs]: More precise
```suggestion
    /// * `msgs` - The `0`-bit messages to use during the oblivious transfer.
```

File: ot/mpz-ot/src/lib.rs [hunk 4]
```diff
@@ -205,6 +254,18 @@ pub trait OTSenderShared<T> {
...
 
+/// A correlated oblivious transfer sender that can be used via a shared reference.
+#[async_trait]
+pub trait COTSenderShared<T> {
+    /// Obliviously transfers correlated messages to the receiver.
+    ///
+    /// # Arguments
+    ///
+    /// * `id` - The unique identifier for this transfer.
+    /// * `msgs` - The `0`-bit messages to obliviously transfer.
```
Comment (themighty1) [inline ot/mpz-ot/src/lib.rs]:
```suggestion
    /// * `msgs` - The `0`-bit messages to use during the oblivious transfer.
```

**Review by themighty1 — APPROVED** [review]

File: ot/mpz-ot/src/ideal/shared/cot.rs (hunk 1 above)
Comment (sinui0) [reply ot/mpz-ot/src/ideal/shared/cot.rs]:
```suggestion
    async fn test_ideal_cot_shared() {
```

**Pull request merged**